python/lsst/sims/coordUtils/LsstCameraUtils.py

```diff
@@ -143,59 +156,59 @@ def _findDetectorsListLSST(pupilPointList, detectorList, allow_multiple_chips=Fa
-                if detector.getName() not in checked_detectors:
-                    checked_detectors.append(detector.getName())
-
-                    # in order to avoid constantly re-instantiating the same afwCameraGeom detector,
-                    # we will now find all of the (RA, Dec) pairs that could be on the present
-                    # chip and test them.
-                    unfound_pts = np.where(chip_has_found < 0)[0]
-                    if len(unfound_pts) == 0:
-                        # we have already found all of the (RA, Dec) pairs
-                        for ix, name in enumerate(outputNameList):
-                            if isinstance(name, list):
-                                outputNameList[ix] = str(name)
-                        return np.array(outputNameList)
-
-                    valid_pt_dexes = np.array([ii for ii in unfound_pts if detector in detectorList[ii]])
-                    if len(valid_pt_dexes) > 0:
-                        valid_pt_list = [nativePointList[ii] for ii in valid_pt_dexes]
-                        transform = detector.getTransform(lsst_camera()._nativeCameraSys, PIXELS)
-                        detectorPointList = transform.applyForward(valid_pt_list)
-
-                        box = afwGeom.Box2D(detector.getBBox())
-                        for ix, pt in zip(valid_pt_dexes, detectorPointList):
-                            if box.contains(pt):
-                                if not could_be_multiple[ix]:
-                                    # because this (RA, Dec) pair is not marked as could_be_multiple,
-                                    # the fact that this (RA, Dec) pair is on the current chip
-                                    # means this (RA, Dec) pair no longer needs to be considered.
-                                    # You can set chip_has_found[ix] to unity.
-                                    outputNameList[ix] = detector.getName()
-                                    chip_has_found[ix] = 1
-                                else:
-                                    # Since this (RA, Dec) pair has been makred could_be_multiple,
-                                    # finding this (RA, Dec) pair on the chip does not remove the
-                                    # (RA, Dec) pair from contention.
-                                    if outputNameList[ix] is None:
-                                        outputNameList[ix] = detector.getName()
-                                    elif isinstance(outputNameList[ix], list):
-                                        outputNameList[ix].append(detector.getName())
-                                    else:
-                                        outputNameList[ix] = [outputNameList[ix], detector.getName()]
+    # t_assemble_list = 0.0
+    # loop over detectors
+    for i_detector, detector in enumerate(detectorList):
+        if len(possible_points[i_detector]) == 0:
+            continue
+
+        if unfound_pts <= 0:
```

Comment:
Why is this <= 0? When would you expect this to be less than 0?

Reply:
No, but I am paranoid that I may accidentally have decremented it one too many times.

That being said, if it does go < 0, that would be a sign that something is wrong.  I will have it throw an exception if that happens